mopidy/core/tracklist.py

```diff
@@ -563,6 +559,26 @@ def slice(self, start, end):
         # TODO: validate slice?
         return self._tl_tracks[start:end]
 
+    def _remove_tracks(self, tl_tracks, *, stop_if_current=True):
+        """Internal method for removing tracks from the tracklist."""
+        # Check to see if we're removing the current tl_track
+        # and stop playback if we are
+        # If we don't do this the track keeps playing and we no longer
+        # have a reference to it
+
+        current_track = self.core.playback.get_current_tl_track()
+        if stop_if_current and current_track in tl_tracks:
+            self.core.playback.stop()
+
+        for tl_track in tl_tracks:
+            # Check in case it got consumed
```

Comment:
Yes. Consider the situation where you're removing a large group of tracks and the current track is at the end of the list. core.playback.stop() can consume that track before the loop gets to it. It's a very hard one to write a test for since it's so timing-dependant - I couldn't provoke it in mock tests on my Core i7 even with 50 tracks in the tracklist, but it's pretty easy to make mopidy do it on a Raspberry Pi in real-life usage.